app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,126 @@
+samson.controller('ProjectRolesCtrl', function ($scope, $element, $filter, projectRolesService, messageCenterService) {
+    $scope.project_role = {};
+    $scope.roles = [];
+
+    $scope.initModel = function () {
+        $scope.project_role = {
+            id: getUniqueId(),
+            user_id: getUserId(),
+            project_id: getProjectId(),
+            role_id: getRoleId()
+        };
+
+        loadProjectRoles();
+    };
+
+    $scope.$watch('project_role.role_id', function (new_role_value, old_role_value) {
+        if (new_role_value !== old_role_value) {
```

Comment:
This function is not meant to return anything. It's meant to trigger a create / update operation against the backend when the new_value doesn't match the old_value. If the values match, it will just be ignored (and they do match when you're initialising the controller, they're both 'undefined' or empty, cannot remember for sure).

Anyway, I think it's a common approach when watching for changes on the model.